android/app/src/main/java/com/zulipmobile/MainApplication.java

```diff
@@ -85,11 +85,12 @@ public IPushNotification getPushNotification(Context context, Bundle bundle, App
             clearConversations(conversations);
             NotificationManager nMgr = (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);
             nMgr.cancelAll();
-            return null;
```

Comment:
Hmm -- this adds a new wrinkle. Why change the behavior for the case of these dismiss intents?

Reply:
To remove all the null return calls. After removing the return call we recieve a stacktrace.

```
08-09 18:03:08.885 3661-3661/? D/LogHelper: MXMNotificationListener - removed notification: com.zulipmobile.debug
```

I read about that stacktrace, and got to know its because the notification doesn't contain a data key which is required by each GCM message, whereas this is a fake one generated by the ProxyService, Very less functions are called for completing the function.
Although return null can also be used.

Reply:
That line you quoted is not a stack trace. :-P A stack trace has a list with a function's caller, and its caller, and so on. Was there a stack trace after it? The actual stack trace would be good to paste if discussing it; I can't quite tell what you're saying.

The case where we enter this `if` body is with a "notification" that is a fake one generated not by the ProxyService, but by our own code. It's the bundle we add to `dismissIntent` in our `getNotificationBuilder`. ProxyService is just the place where we have chosen, in that same block of our own code, to point that intent at. It receives it, pulls out that bundle, and invokes this function.

Reply:
Haha yeah it's not a stack trace, I meant a debug log message.